stdpopsim/cli.py

```diff
@@ -348,26 +349,52 @@ def add_simulate_species_parser(parser, species):
 
     if len(species.genome.chromosomes) > 1:
         choices = [chrom.id for chrom in species.genome.chromosomes]
+        choices.append("any")
         species_parser.add_argument(
             "-c", "--chromosome", choices=choices, metavar="", default=choices[0],
             help=(
                 f"Simulate a specific chromosome. "
                 f"Options: {', '.join(choices)}. "
                 f"Default={choices[0]}."))
     species_parser.add_argument(
-        "-l", "--length-multiplier", default=1, type=float,
+        "-l", "--length-multiplier", default=1, type=float, metavar="l",
         help="Simulate a sequence of length l times the named chromosome's length, "
              "using the named chromosome's mutation and recombination rates.")
+
+    def length(arg, parser=species_parser):
+        x = 1
+        if arg[-1] == "k":
+            arg, x = arg[:-1], 1000
+        elif arg[-1] == "m":
+            arg, x = arg[:-1], 1000000
+        try:
+            arg = int(arg)
+        except ValueError:
+            parser.error(f"Invalid length: {arg}")
+        return x * arg
+    species_parser.add_argument(
+        "-L", "--length", type=length, metavar="L", default=None,
+        help="Simulate a sequence of length L, "
+             "using the named chromosome's mutation and recombination rates. "
+             "The suffixes `k` or `m` are permitted, which indicate kilobases "
+             "and megabases respectively. "
+             "If no genomic coordinate is specified, it will be drawn randomly. "
+             "If the special chromosome `any` is requested, a chromosome will "
+             "also be drawn at random, with probability proportional to the "
+             "chromosome length.")
+    species_parser.add_argument(
```

Comment:
Hmmm. There's a lot of good stuff in here, but I wonder if having a ``--start`` and ``--end`` coordinate as an argument would be simpler. Since people are going to be using this to slice up the genome into N bits, specifying a single slice is the most basic way of doing this, which someone can build more sophisticated stuff on top of later. The user already needs to know how long the chromosome is.

I would vote for just having a simple ``start`` and ``end`` coordinates specified as the first pass, and we could think about having things like ``--end=+1M`` later to make specifying less precise slices a bit easier.

Reply:
We could also add things like ``--start=uniform`` or something later too, if we like.